packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/mcp/search_handler.py

```
@@ -460,4 +470,209 @@ def _apply_attachment_filters(
+        return filtered_results
+
+    def _apply_lightweight_attachment_filters(
+        self, results: list[HybridSearchResult], attachment_filter: dict[str, Any]
+    ) -> list[HybridSearchResult]:
+        """Fast filtering optimized for attachment discovery across all sources."""
+        filtered_results = []
+        
+        for result in results:
+            # Quick attachment detection - avoid expensive checks
+            is_attachment = (
+                result.is_attachment or 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)